stl/inc/format

```diff
@@ -611,17 +696,35 @@ _NODISCARD constexpr const _CharT* _Parse_replacement_field(
     return _Begin + 1;
 }
 
+template <class _CharT>
+const _CharT* _Scan_format_string(const _CharT* _First, const _CharT* _Last, const _CharT _Val, const _Cvtvec& _Cvt) {
+    if (_Cvt._Mbcurmax == 1 || _Cvt._Mbcurmax == 4) {
+        return _Find_unchecked(_First, _Last, _Val);
+    }
+
+    while (_First != _Last && *_First != _Val) {
+        const int _Units = _Code_units_in_next_character(_First, _Last, _Cvt);
+        if (_Units < 0) {
+            throw format_error("Invalid encoded character in format string.");
+        }
+        _First += _Units;
+    }
+
+    return _First;
+}
+
 template <class _CharT, _Parse_replacement_field_callbacks<_CharT> _HandlerT>
-constexpr void _Parse_format_string(basic_string_view<_CharT> _Format_str, _HandlerT&& _Handler) {
-    auto _Begin = _Format_str.data();
-    auto _End   = _Begin + _Format_str.size();
+void _Parse_format_string(basic_string_view<_CharT> _Format_str, _HandlerT&& _Handler) {
+    auto _Begin         = _Format_str.data();
+    auto _End           = _Begin + _Format_str.size();
+    const _Cvtvec& _Cvt = ::_Getcvt();
+
     while (_Begin != _End) {
-        const _CharT* _OpeningCurl = _Begin;
-        if (*_Begin != '{') {
-            // we didn't start at an opening curl, find the next one
-            _OpeningCurl = _Find_unchecked(_Begin + 1, _End, '{');
+        const _CharT* _OpeningCurl = _Scan_format_string(_Begin, _End, _CharT{'{'}, _Cvt);
+        if (_Begin != _OpeningCurl) {
+            // we didn't start at an opening curl

```

Comment:
@barcharcraz, Pay special attention to the sneaky logic change here which avoids needing to increment `_Begin` past an encoded character (i.e., this is the other broken `+ 1`).

Reply:
This removes a micro-optimization (don't call `find` if we started at an opening curl). The comment no longer makes sense.

Reply:
This comment has never been useful, I'll just strike it. Restoring the micro-optimization is problematic, since skipping an arbitrary encoded character is more expensive than the previous `_Begin + 1` which resulted in #1576. I suspect `_Begin + _Code_units_in_next_character(/**/)` isn't a significant win. Maybe:
```c++
const _CharT* _OpeningCurl = _Begin;
if (*_Begin != `{`) {
    _OpeningCurl = _Scan_format_string(_Begin, _End, _CharT{'{'}, _Cvt);
```
which redundantly checks `*_Begin` in the branch would still be an improvement? I'll apply something along these lines in the morning.